pkg/devfile/components.go

```diff
@@ -0,0 +1,265 @@
+package devfile
+
+import (
+	"fmt"
+	"net/url"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/devfile/api/v2/pkg/apis/workspaces/v1alpha2"
+	devfilev1 "github.com/devfile/api/v2/pkg/apis/workspaces/v1alpha2"
+	"github.com/devfile/library/pkg/devfile/parser"
+	"github.com/devfile/library/pkg/devfile/parser/data"
+	"github.com/devfile/library/pkg/devfile/parser/data/v2/common"
+	parsercommon "github.com/devfile/library/pkg/devfile/parser/data/v2/common"
+	devfileFileSystem "github.com/devfile/library/pkg/testingutil/filesystem"
+	devfilefs "github.com/devfile/library/pkg/testingutil/filesystem"
+)
+
+const (
+	UriFolder  = "kubernetes"
+	filePrefix = "odo-service-"
+)
+
+type InlinedComponent struct {
+	Name    string
+	Inlined string
+}
+
+type URIComponent struct {
+	Name string
+	URI  string
+}
+
+// SetupTestFolder for testing
+func SetupTestFolder(testFolderName string, fs devfileFileSystem.Filesystem) (devfileFileSystem.File, error) {
```

Comment:
Since this function is for testing, why not put it in the `pkg/devfile/components_test.go` file where it is being used?

Reply:
Because it is used in another package (pkg/service) and functions in _test.go files are not exported. I created a `testing` package instead, to make clear this is a test-related function